src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,245 @@
+  };
+
+  const fetchSelectedOption = useCallback(
+    async (selectionValue: string) => {
+      const res = await fetch(
+        `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+      );
+      const { custom_object_record } = await res.json();
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+      onChange(custom_object_record.id);
+    },
+    [customObjectKey, onChange]
+  );
+
+  const handleChange = useCallback<NonNullable<IComboboxProps["onChange"]>>(
+    async ({ inputValue, selectionValue }) => {
+      if (selectionValue !== undefined) {
+        if (selectionValue == "") {
+          setSelectedOption(EMPTY_OPTION);
+          setInputValue(EMPTY_OPTION.name);
+          setOptions([]);
+          onChange(EMPTY_OPTION.value);
+        } else {
+          const selectedOption = options.find(
+            (option) => option.value === selectionValue
+          );
+          if (selectedOption) {
+            setSelectedOption(selectedOption);
+            setInputValue(selectedOption.name);
+            onChange(selectedOption.value);
+            setOptions([selectedOption]);
+          }
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        const searchParams = new URLSearchParams();
+        searchParams.set("name", inputValue.toLocaleLowerCase());
+        searchParams.set("source", "zen:ticket");
+        searchParams.set("field_id", fieldId.toString());
+        searchParams.set("requester_id", userId.toString());
+        if (organizationId !== null)
+          searchParams.set("organization_id", organizationId);
+        setIsLoadingOptions(true);
+        try {
+          const response = await fetch(
+            `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+          );
+
+          const data = await response.json();
+          if (data !== undefined) {
+            setOptions(
+              data.custom_object_records.map(
+                ({ name, id }: { name: string; id: string }) => ({
+                  name,
+                  value: id,
+                })
+              )
+            );
+          } else {
+            setOptions([]);
+          }
+        } catch (error) {
+          return error;
+        } finally {
+          setIsLoadingOptions(false);
+        }
+      }
+      return;
+    },
+    [customObjectKey, userId, organizationId, fieldId, onChange, options]
+  );
+
+  const debounceHandleChange = useMemo(
+    () => debounce(handleChange, 300),
+    [handleChange]
+  );
+
+  useEffect(() => {
+    return () => debounceHandleChange.cancel();
+  }, [debounceHandleChange]);
+
+  useEffect(() => {
+    if (value) {
+      fetchSelectedOption(value as string);
+    }
+  }, []);
```

Comment:
I would add a comment here explaining why we set the dependencies array to `[]`, to avoid the possibility of somebody changing it in the future

Reply:
ok, I added comment